### PR TITLE
 feat: add custom openapi template 

### PR DESCRIPTION
**Pull request by beeman**
This PR vendors in the latest openapi-template from [here](https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator/src/main/resources/kotlin-client) and adds support for sending along headers with the request

It also [tweaks the Gradle setup](https://github.com/kin-labs/kinetic-android-sdk/commit/0f857033fe8d90285809061cb1f470a43c122871) to automatically download the latest Swagger file, as this generator [does not support remote locations for inputspec](https://github.com/OpenAPITools/openapi-generator/issues/8255)

### CODE REVIEW

File: app/src/main/java/org/kin/kineticdemo/MainActivity.kt [hunk 1]
```diff
@@ -42,9 +42,10 @@ class MainActivity : AppCompatActivity() {
 
         kineticNetworkScope.launch {
             kinetic = KineticSdk.setup(
-                "http://staging.kinetic.host",
-                "devnet",
-                1
+                endpoint = "https://sandbox.kinetic.host",
+                environment = "devnet",
+                index = 1,
+                headers = mapOf("kinetic-custom-header" to "Yay nice!"),
```
Comment (samdowd) [inline app/src/main/java/org/kin/kineticdemo/MainActivity.kt]: Was it intentional adding the `paramName = ` in places or is that copilot? Generally we leave them out on Android as the IDE shows them for you.

File: kinetic/src/main/java/org/kin/kinetic/KineticSdkInternal.kt [hunk 2]
```diff
@@ -43,12 +44,12 @@ class KineticSdkInternal(
         apiHeaders += headers
         apiHeaders.put("kinetic-environment", environment)
         apiHeaders.put("kinetic-index", index.toString())
-        apiHeaders.put("kinetic-user-agent", "Kinetic Android SDK v0.0.1-rc.1")
+        apiHeaders.put("kinetic-user-agent", "Kinetic Android SDK v1.0.0-rc.3")
```
Comment (samdowd) [inline kinetic/src/main/java/org/kin/kinetic/KineticSdkInternal.kt]: I have a constants file with the name and version. I'll use those values here

**Review by samdowd — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/kin/kineticdemo/MainActivity.kt (hunk 1 above)
Comment (beeman) [reply app/src/main/java/org/kin/kineticdemo/MainActivity.kt]: Copilot, of course... 😇 I removed them.

File: kinetic/src/main/java/org/kin/kinetic/KineticSdkInternal.kt (hunk 2 above)
Comment (beeman) [reply kinetic/src/main/java/org/kin/kinetic/KineticSdkInternal.kt]: Nice! And it looks like you also had a helper method defined like in the TS SDK, which it's now using.

**Review by samdowd — APPROVED** [review]
Looks great!

**Pull request merged**